ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/search_tree/algorithms/SearchTree.java

```diff
@@ -246,9 +252,10 @@ private void iterateOnTree() {
      */
     private void updateOptimalLeafWithNextDepthBestLeaf(AbstractNetworkPool networkPool) throws InterruptedException {
 
-        final List<NetworkActionCombination> naCombinations = bloomer.bloom(optimalLeaf, input.getOptimizationPerimeter().getNetworkActions());
+        final Map<NetworkActionCombination, Boolean> naCombinations = bloomer.bloom(optimalLeaf, input.getOptimizationPerimeter().getNetworkActions());
```

Comment:
I'm not sure we should do this because right after the code you highlighted we do this : 

` naCombinations.sort(this::deterministicNetworkActionCombinationComparison);`

It was done before our development to sort combinations a certain way. So I do not see why using a SortedMap could be useful if we sort its keys a different way immediately after.

Reply:
yes, i meant couldn't you handle one object (a sorted map), rather than having to create a second object (the list ?) 
if you had a sorted map you could directly use the keyset